staging/src/k8s.io/kubectl/pkg/cmd/portforward/portforward.go

```diff
@@ -393,7 +393,16 @@ func (o PortForwardOptions) RunPortForward() error {
 	}
 
 	if pod.Status.Phase != corev1.PodRunning {
-		return fmt.Errorf("unable to forward port because pod is not running. Current status=%v", pod.Status.Phase)
+		return fmt.Errorf("unable to forward port because pod phase is not running. Current phase=%v", pod.Status.Phase)
+	}
+
+	readyCondition, err := polymorphichelpers.GetPodCondition(pod, corev1.PodReady)
+	if err != nil {
+		return fmt.Errorf("unable to fetch Ready condition for pod %s: %w", pod.Name, err)
+	}
+
+	if readyCondition.Status == corev1.ConditionFalse {
```

Comment:
if ConditionUknown or missing we send traffic to the pod?

Reply:
I wrote `GetPodCondition` to return an error if the requested condition is not found so this will hard fail if the pod is not condition Ready == True.

Reply:
If the condition is found, but its value is "Unknown", I think it would pass.

For example, here the Ready condition is found, but its status is Unknown:
```
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2023-05-02T03:24:46Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2023-05-02T03:24:46Z"
    status: "Unknown"
    type: Ready
```

According to this comment, I guess it is possible, and also possible there could be other conditions in the future, although I think that is unlikely.
```
// These are valid condition statuses. "ConditionTrue" means a resource is in the condition.
// "ConditionFalse" means a resource is not in the condition. "ConditionUnknown" means kubernetes
// can't decide if a resource is in the condition or not. In the future, we could add other
// intermediate conditions, e.g. ConditionDegraded.
const (
	ConditionTrue    ConditionStatus = "True"
	ConditionFalse   ConditionStatus = "False"
	ConditionUnknown ConditionStatus = "Unknown"
)
```

It comes down to how it should handle unknown, optimistically (attempt to port forward) or pessimistically (fail).